lib/installation-proxy/index.js

```diff
@@ -0,0 +1,56 @@
+const INSTALLATION_PROXY_SERVICE_NAME = 'com.apple.mobile.installation_proxy';
+
+class InstallationProxyService {
+  constructor (plistService) {
+    this.plistService = plistService;
+  }
+
+  async listApplications (applicationType = 'User') {
+    const request = {
+      Command: 'Browse',
+      ClientOptions: {
+        ApplicationType: applicationType
+      }
+    };
+
+    this.plistService.sendPlist(request);
+    const messages = await this.waitMessageCompletion();
+    let map = {};
```

Comment:
consider using messages.reduce

Reply:
I had an experiment. Is this what you meant?
```javascript
return messages.reduce((acc, message) => {
      message.CurrentList.forEach(app => {
        acc[app.CFBundleIdentifier] = app;
      });
      return acc;
    }, {});
```

Reply:
yep, better